internal/gcsx/file_cache_reader.go

```diff
@@ -50,8 +51,14 @@ type FileCacheReader struct {
 	// will be downloaded for random reads as well too.
 	cacheFileForRangeRead bool
 
+	// mu protects fileCacheHandle from concurrent access across multiple threads
```

Comment:
Thanks for catching this!!

Taken care as part of this PR: https://github.com/GoogleCloudPlatform/gcsfuse/pull/4016, need to be rebased.